tree/ntuple/v7/test/rfield_vector.cxx

```diff
@@ -31,6 +31,51 @@ TEST(RNTuple, ClassVector)
    }
 }
 
+
+TEST(RNTuple, InsideCollection)
+{
+   FileRaii fileGuard("test_ntuple_insidecollection.root");
+
+   auto modelWrite = RNTupleModel::Create();
+   auto wrKlassVec = modelWrite->MakeField<std::vector<CustomStruct>>("klassVec");
+   CustomStruct klass;
+   klass.a = 42.0;
+   klass.v1.emplace_back(1.0);
+   klass.v1.emplace_back(2.0);
+   wrKlassVec->emplace_back(klass);
+
+   {
+      RNTupleWriter ntuple(std::move(modelWrite),
+         std::make_unique<RPageSinkFile>("myNTuple", fileGuard.GetPath(), RNTupleWriteOptions()));
+      ntuple.Fill();
+   }
+
+   auto source = std::make_unique<RPageSourceFile>("myNTuple", fileGuard.GetPath(), RNTupleReadOptions());
+   source->Attach();
+   EXPECT_EQ(1U, source->GetNEntries());
+
+   auto idKlassVec = source->GetDescriptor().FindFieldId("klassVec");
+   ASSERT_NE(idKlassVec, ROOT::Experimental::kInvalidDescriptorId);
+   auto idKlass = source->GetDescriptor().FindFieldId("CustomStruct", idKlassVec);
+   ASSERT_NE(idKlass, ROOT::Experimental::kInvalidDescriptorId);
+   auto idA = source->GetDescriptor().FindFieldId("a", idKlass);
+   ASSERT_NE(idA, ROOT::Experimental::kInvalidDescriptorId);
+   auto fieldInner = std::unique_ptr<RFieldBase>(RFieldBase::Create("klassVec.a", "float"));
+   RFieldFuse::ConnectRecursively(idA, *source, *fieldInner);
+
+   auto field = std::make_unique<ROOT::Experimental::RVectorField>("klassVec", std::move(fieldInner));
+   RFieldFuse::Connect(idKlassVec, *source, *field);
```

Comment:
I don't understand the API here:

- why do we need the last two lines? Couldn't I read `klassVec.a` using `fieldInner`?
- why does `"klassVec.a"`, a float, need `ConnectRecursively` and `"klassVec"`, a complex object, is ok with just `Connect`?

Maybe what I'm missing is what `RFieldFuse` is and does...I could not find docs though.

Reply:
Unless `RFieldFuse::Connect` has been called, an `RField` object is dangling and not attached to any concrete storage backend.  The `RFieldFuse::Connect` call binds the columns that back the field to an underlying storage. Doing so recursively means for the field itself and all its subfields.  Here. we don't want to connect the vector of `klassVec` but only the vector of `klassVec.a`. A vector is always an outer field (for the length of the vector) plus a tree of inner fields (for the vector type). In line 64, we could as well use `RFieldFuse::Connect` instead of `RFieldFuse::ConnectRecursively` but only because `klassVec.a` is a simple type with no sub fields.